{"relationshipsSections":[{"identifiers":["doc:\/\/swift-argument-parser.ArgumentParser\/SQ","doc:\/\/swift-argument-parser.ArgumentParser\/s5ErrorP","doc:\/\/swift-argument-parser.ArgumentParser\/SH","doc:\/\/swift-argument-parser.ArgumentParser\/SY","doc:\/\/swift-argument-parser.ArgumentParser\/s8SendableP"],"title":"Conforms To","kind":"relationships","type":"conformsTo"}],"primaryContentSections":[{"kind":"declarations","declarations":[{"platforms":["macOS"],"languages":["swift"],"tokens":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"text":"ExitCode","kind":"identifier"}]}]},{"content":[{"anchor":"overview","level":2,"text":"Overview","type":"heading"},{"inlineContent":[{"text":"If youâ€™re printing custom error messages yourself, you can throw this error","type":"text"},{"text":" ","type":"text"},{"type":"text","text":"to specify the exit code without adding any additional output to standard"},{"text":" ","type":"text"},{"text":"out or standard error.","type":"text"}],"type":"paragraph"}],"kind":"content"}],"variants":[{"traits":[{"interfaceLanguage":"swift"}],"paths":["\/documentation\/argumentparser\/exitcode"]}],"seeAlsoSections":[{"generated":true,"anchor":"Validation-and-Errors","title":"Validation and Errors","identifiers":["doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/Validation","doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ValidationError","doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/CleanExit"]}],"schemaVersion":{"minor":3,"major":0,"patch":0},"kind":"symbol","identifier":{"url":"doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ExitCode","interfaceLanguage":"swift"},"sections":[],"metadata":{"title":"ExitCode","modules":[{"name":"ArgumentParser"}],"roleHeading":"Structure","navigatorTitle":[{"kind":"identifier","text":"ExitCode"}],"role":"symbol","fragments":[{"text":"struct","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"ExitCode"}],"symbolKind":"struct","externalID":"s:14ArgumentParser8ExitCodeV"},"abstract":[{"type":"text","text":"An error type that only includes an exit code."}],"hierarchy":{"paths":[["doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser"]]},"topicSections":[{"generated":true,"identifiers":["doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ExitCode\/init(_:)","doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ExitCode\/init(rawValue:)"],"title":"Initializers","anchor":"Initializers"},{"identifiers":["doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ExitCode\/isSuccess","doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ExitCode\/rawValue"],"generated":true,"anchor":"Instance-Properties","title":"Instance Properties"},{"generated":true,"identifiers":["doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ExitCode\/failure","doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ExitCode\/success","doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ExitCode\/validationFailure"],"title":"Type Properties","anchor":"Type-Properties"},{"title":"Default Implementations","generated":true,"anchor":"Default-Implementations","identifiers":["doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ExitCode\/Equatable-Implementations","doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ExitCode\/Error-Implementations","doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ExitCode\/RawRepresentable-Implementations"]}],"references":{"doc://swift-argument-parser.ArgumentParser/SQ":{"identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/SQ","title":"Swift.Equatable","type":"unresolvable"},"doc://swift-argument-parser.ArgumentParser/documentation/ArgumentParser/ExitCode/Equatable-Implementations":{"kind":"article","abstract":[],"identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ExitCode\/Equatable-Implementations","url":"\/documentation\/argumentparser\/exitcode\/equatable-implementations","title":"Equatable Implementations","type":"topic","role":"collectionGroup"},"doc://swift-argument-parser.ArgumentParser/documentation/ArgumentParser/ExitCode/init(rawValue:)":{"kind":"symbol","title":"init(rawValue:)","type":"topic","abstract":[],"url":"\/documentation\/argumentparser\/exitcode\/init(rawvalue:)","identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ExitCode\/init(rawValue:)","role":"symbol","fragments":[{"text":"init","kind":"identifier"},{"kind":"text","text":"("},{"text":"rawValue","kind":"externalParam"},{"kind":"text","text":": "},{"preciseIdentifier":"s:s5Int32V","kind":"typeIdentifier","text":"Int32"},{"text":")","kind":"text"}]},"doc://swift-argument-parser.ArgumentParser/SY":{"type":"unresolvable","title":"Swift.RawRepresentable","identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/SY"},"doc://swift-argument-parser.ArgumentParser/s8SendableP":{"type":"unresolvable","title":"Swift.Sendable","identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/s8SendableP"},"doc://swift-argument-parser.ArgumentParser/documentation/ArgumentParser/ExitCode/RawRepresentable-Implementations":{"kind":"article","title":"RawRepresentable Implementations","url":"\/documentation\/argumentparser\/exitcode\/rawrepresentable-implementations","identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ExitCode\/RawRepresentable-Implementations","type":"topic","abstract":[],"role":"collectionGroup"},"doc://swift-argument-parser.ArgumentParser/documentation/ArgumentParser/CleanExit":{"navigatorTitle":[{"text":"CleanExit","kind":"identifier"}],"abstract":[{"type":"text","text":"An error type that represents a clean (i.e. non-error state) exit of the"},{"text":" ","type":"text"},{"type":"text","text":"utility."}],"type":"topic","title":"CleanExit","identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/CleanExit","url":"\/documentation\/argumentparser\/cleanexit","kind":"symbol","role":"symbol","fragments":[{"kind":"keyword","text":"struct"},{"text":" ","kind":"text"},{"text":"CleanExit","kind":"identifier"}]},"doc://swift-argument-parser.ArgumentParser/documentation/ArgumentParser/ExitCode/isSuccess":{"kind":"symbol","fragments":[{"text":"var","kind":"keyword"},{"kind":"text","text":" "},{"text":"isSuccess","kind":"identifier"},{"text":": ","kind":"text"},{"preciseIdentifier":"s:Sb","text":"Bool","kind":"typeIdentifier"}],"abstract":[{"type":"text","text":"A Boolean value indicating whether this exit code represents the"},{"type":"text","text":" "},{"type":"text","text":"successful completion of a command."}],"identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ExitCode\/isSuccess","url":"\/documentation\/argumentparser\/exitcode\/issuccess","title":"isSuccess","type":"topic","role":"symbol"},"doc://swift-argument-parser.ArgumentParser/documentation/ArgumentParser/ExitCode":{"abstract":[{"text":"An error type that only includes an exit code.","type":"text"}],"type":"topic","identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ExitCode","title":"ExitCode","role":"symbol","url":"\/documentation\/argumentparser\/exitcode","navigatorTitle":[{"kind":"identifier","text":"ExitCode"}],"fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"text":"ExitCode","kind":"identifier"}],"kind":"symbol"},"doc://swift-argument-parser.ArgumentParser/documentation/ArgumentParser/ExitCode/failure":{"kind":"symbol","title":"failure","type":"topic","abstract":[{"type":"text","text":"An exit code that indicates that the command failed."}],"url":"\/documentation\/argumentparser\/exitcode\/failure","identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ExitCode\/failure","role":"symbol","fragments":[{"text":"static","kind":"keyword"},{"kind":"text","text":" "},{"text":"let","kind":"keyword"},{"text":" ","kind":"text"},{"text":"failure","kind":"identifier"},{"kind":"text","text":": "},{"text":"ExitCode","kind":"typeIdentifier","preciseIdentifier":"s:14ArgumentParser8ExitCodeV"}]},"doc://swift-argument-parser.ArgumentParser/documentation/ArgumentParser/ExitCode/validationFailure":{"kind":"symbol","fragments":[{"kind":"keyword","text":"static"},{"text":" ","kind":"text"},{"text":"let","kind":"keyword"},{"text":" ","kind":"text"},{"kind":"identifier","text":"validationFailure"},{"text":": ","kind":"text"},{"kind":"typeIdentifier","preciseIdentifier":"s:14ArgumentParser8ExitCodeV","text":"ExitCode"}],"abstract":[{"text":"An exit code that indicates that the user provided invalid input.","type":"text"}],"identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ExitCode\/validationFailure","url":"\/documentation\/argumentparser\/exitcode\/validationfailure","title":"validationFailure","type":"topic","role":"symbol"},"doc://swift-argument-parser.ArgumentParser/documentation/ArgumentParser":{"title":"ArgumentParser","type":"topic","abstract":[{"text":"Straightforward, type-safe argument parsing for Swift.","type":"text"}],"identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser","role":"collection","url":"\/documentation\/argumentparser","kind":"symbol"},"doc://swift-argument-parser.ArgumentParser/documentation/ArgumentParser/ExitCode/success":{"fragments":[{"kind":"keyword","text":"static"},{"text":" ","kind":"text"},{"text":"let","kind":"keyword"},{"kind":"text","text":" "},{"text":"success","kind":"identifier"},{"kind":"text","text":": "},{"kind":"typeIdentifier","preciseIdentifier":"s:14ArgumentParser8ExitCodeV","text":"ExitCode"}],"kind":"symbol","abstract":[{"text":"An exit code that indicates successful completion of a command.","type":"text"}],"url":"\/documentation\/argumentparser\/exitcode\/success","title":"success","identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ExitCode\/success","type":"topic","role":"symbol"},"doc://swift-argument-parser.ArgumentParser/s5ErrorP":{"title":"Swift.Error","type":"unresolvable","identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/s5ErrorP"},"doc://swift-argument-parser.ArgumentParser/documentation/ArgumentParser/Validation":{"kind":"article","title":"Providing Custom Validation","url":"\/documentation\/argumentparser\/validation","abstract":[{"type":"text","text":"Provide helpful feedback to users when things go wrong."}],"identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/Validation","role":"article","type":"topic"},"doc://swift-argument-parser.ArgumentParser/documentation/ArgumentParser/ValidationError":{"identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ValidationError","type":"topic","navigatorTitle":[{"kind":"identifier","text":"ValidationError"}],"fragments":[{"text":"struct","kind":"keyword"},{"text":" ","kind":"text"},{"text":"ValidationError","kind":"identifier"}],"url":"\/documentation\/argumentparser\/validationerror","kind":"symbol","role":"symbol","abstract":[{"type":"text","text":"An error type that is presented to the user as an error with parsing their"},{"text":" ","type":"text"},{"type":"text","text":"command-line input."}],"title":"ValidationError"},"doc://swift-argument-parser.ArgumentParser/documentation/ArgumentParser/ExitCode/Error-Implementations":{"kind":"article","abstract":[],"identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ExitCode\/Error-Implementations","url":"\/documentation\/argumentparser\/exitcode\/error-implementations","title":"Error Implementations","type":"topic","role":"collectionGroup"},"doc://swift-argument-parser.ArgumentParser/documentation/ArgumentParser/ExitCode/init(_:)":{"kind":"symbol","title":"init(_:)","type":"topic","abstract":[{"text":"Creates a new ","type":"text"},{"code":"ExitCode","type":"codeVoice"},{"text":" with the given code.","type":"text"}],"url":"\/documentation\/argumentparser\/exitcode\/init(_:)","identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ExitCode\/init(_:)","role":"symbol","fragments":[{"text":"init","kind":"identifier"},{"text":"(","kind":"text"},{"preciseIdentifier":"s:s5Int32V","kind":"typeIdentifier","text":"Int32"},{"kind":"text","text":")"}]},"doc://swift-argument-parser.ArgumentParser/SH":{"title":"Swift.Hashable","type":"unresolvable","identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/SH"},"doc://swift-argument-parser.ArgumentParser/documentation/ArgumentParser/ExitCode/rawValue":{"type":"topic","kind":"symbol","abstract":[{"type":"text","text":"The exit code represented by this instance."}],"url":"\/documentation\/argumentparser\/exitcode\/rawvalue","fragments":[{"text":"var","kind":"keyword"},{"kind":"text","text":" "},{"kind":"identifier","text":"rawValue"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"Int32","preciseIdentifier":"s:s5Int32V"}],"identifier":"doc:\/\/swift-argument-parser.ArgumentParser\/documentation\/ArgumentParser\/ExitCode\/rawValue","role":"symbol","title":"rawValue"}}}